components/admin_console/filter/filter.tsx

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import FilterList from './filter_list';
+import './filter.scss';
+
+export type Filters = {
+    [filterKey: string]: string[];
+};
+
+export type FilterValue = {
+    name: string | JSX.Element;
+    value: boolean | string | string[];
+};
+
+export type FilterValues = {
+    [key: string]: FilterValue;
+};
+
+export type FilterOption = {
+
+    // Display name of the filter option eg. 'Channels', 'Roles' or <FormattedMessage .../>
+    name: string | JSX.Element;
+
+    // List of keys that match the filter values, used to define the order in which the filters appear
+    keys: string[];
+
+    // Key value map of filter values with keys matching the keys above
+    values: FilterValues;
+
+    // Filter Component type, optional parameter defaults to FilterCheckbox
+    type?: React.ElementType;
+}
+
+export type FilterOptions = {
+    [key: string]: FilterOption;
+}
+
+type Props = {
+    onFilter: (filters: FilterOptions) => void;
+    options: FilterOptions;
+    keys: string[];
+}
+
+type State = {
+    show: boolean;
+    options: FilterOptions;
+    keys: string[];
+    optionsModified: boolean;
+    filterCount: number;
+}
+
+class Filter extends React.PureComponent<Props, State> {
+    private buttonRef: React.RefObject<HTMLButtonElement>;
+    private filterRef: React.RefObject<HTMLDivElement>;
+
+    public constructor(props: Props) {
+        super(props);
+
+        let options = props.options;
+        let keys = props.keys;
+        let error = '';
+        keys.forEach((key) => {
+            const option = options[key];
+            if (option) {
+                option.keys.forEach((optionKey) => {
+                    if (!option.values[optionKey]) {
+                        error = `Invalid Filter option key: ${optionKey}, no matching value for given option key`;
+                    }
+                });
+            } else {
+                error = `Invalid Filter key: ${key}, no matching option for given key`;
```

Comment:
Are these errors OK not to be translated?

Reply:
Actually @mkraft I initially added these as error handling for the component itself, cause the list of keys need to match the options map given (or else the page fails to render) but theres no way to enforce that, so i opted to set the options to empty and log this error

but then removed the logging of the error later (due to linting)... I think ill just remove these messages and have the component fail silently without breaking the page